VERSIONING.md

```diff
@@ -0,0 +1,244 @@
+OpenTelemetry.Instrumentation.AspNetCore must be also split into Traces and
+Metrics ones separately.
+
+Another downside is the lack of "clean transition", arising from the fact that
+there are common classes used to initialize various signals. These
+functionalities are currently in the `Sdk` class of the main package. eg:
+`Sdk.CreateTracerProviderBuilder()`, `Sdk.CreateMeterProviderBuilder()` etc. If
+metrics are in a separate package, the `Sdk` class cannot provide these entry
+points until the metrics code is moved to the main package. So users who use the
+experimental packages, will have to do code changes, when they upgrade to the
+main package consisting of the metrics.
+
+### Example with Approach2
+
+Following shows an example on how the `OpenTelemetry` package versioning works
+with this approach:
+
+`OpenTelemetry` 0.7.0-beta1 release : Pre-release, no API guarantees.
+
+`OpenTelemetry` 1.0.0-RC1 release : Pre-release, no API guarantees, but more
+stable than beta.
+
+`OpenTelemetry` 1.0.0 release : Stable release consisting of only stable signals
+: Traces, Propagators, Baggage.
+
+`OpenTelemetry.Metrics` 1.0.0-alpha release : Pre-release consisting of Metric
+SDK. Alpha indicates early stages of development. Metrics entry point will be
+contained in this package. (eg: new MeterProviderBuild().Build())
+
+`OpenTelemetry` 1.0.1 release : Bug fixes.
+
+`OpenTelemetry` 1.1.0 release : New features added. No metric code yet.
+
+`OpenTelemetry.Metrics` 1.0.0-beta release : Metric evolves to beta status.
+Still a pre-release.
+
+`OpenTelemetry.Metrics` 1.0.0-RC release : Metric evolves to RC status. Still a
+pre-release, but a final RC release will be done when Metrics is declared
+stable.
+
+`OpenTelemetry` 1.2.0 release : Add metric support. This will be additive
+changes. Metrics entry point will be added the main package (just like Traces.)
+(eg: Sdk.CreateMeterProviderBuilder().Build()) User who were previously using
+the *final* RC package of `OpenTelemetry.Metrics`, will need to change the
+Metric entry point code. Apart from that, they'll simply remove
+`OpenTelemetry.Metrics` and update `OpenTelemetry` to 1.2.0, with no *other*
+code changes.
+
+In the case of `OpenTelemetry.API.Metrics` package, users will require code
+changes, as the plan for Metrics API is to have it come from the .NET Runtime.
+
+`OpenTelemetry` 1.3.0 release : More features.
+
+`OpenTelemetry` 2.0.0 release : Remove any Obsolete methods from 1.* releases
+
+### Approach 3 - Same package name, but version differently
+
+The approach keeps the same amount of packages, but release (multiple) versions
+of the same, with experimental signals part of non-stable versions.
+
+This involves managing more than one branch in Github - master for regular work,
+and a separate experimental branch for experimental features. There can be 'n'
+experimental branches, if there is a need. The immediate future will have
+metrics only as separate branch.
```

Comment:
The instrumentations that want to depend on metrics would need to follow the same approach with a separate branch too, right? That seems ok!

Reply:
yea. Otel.inst.AspNet 1.0.0 (from master) will only instrument traces. The same package, version 1.5.0(from metrics branch) will add metrics support as well.